subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/suite/filter/FilterFactoryImpl.kt

```diff
@@ -169,12 +169,30 @@ private class FilterFactoryImpl(
     }
 
     private fun MutableList<TestsFilter>.addImpactAnalysisFilter() {
-        if (impactAnalysisResult != null && impactAnalysisResult.exists()) {
-            val testNames = impactAnalysisResult.readLines()
+        return when (impactAnalysisResult.policy) {
+            is ImpactAnalysisPolicy.Off -> {
+                // do nothing
+            }
+            is ImpactAnalysisPolicy.On.RunAffectedTests -> {
+                addImpactTests(impactAnalysisResult.affectedTests)
+                removeImpactTests(impactAnalysisResult.addedTests)
+                removeImpactTests(impactAnalysisResult.modifiedTests)
```

Comment:
Why did you choose that this removing is the responsibility of filters?
Let's imagine multiple clients of impact analysis' results. Whill they duplicate this logic?

Reply:
that is good catch; i didn't think of it that way

Reply:
Decided to improve this part later